app/src/main/java/org/wikipedia/main/MainFragment.java

```diff
@@ -512,10 +645,47 @@ private void closeSearchFragment(@NonNull SearchFragment fragment) {
         getChildFragmentManager().beginTransaction().remove(fragment).commitNowAllowingStateLoss();
     }
 
+    public void switchToSearchFragment(ImageSearchFragment fragment,
```

Comment:
Could you explain when this fragment would be called and clarify how it would work?

Reply:
This fragment is called when the user is using the Advanced Image Search, where selecting a label opens up the `SearchFragment`. Since the `ImageSearchFragment` is a child fragment of `MainFragment`, the `ImageSearchFragment` has to be closed before opening the `SearchFragment`, which is what `switchToSearchFragment` does. When the `ImageSearchFragment` doesn't get closed, the `SearchFragment` doesn't work properly because it thinks that the `ImageSearchFragment` is its parent, and thus some callback functions don't work as expected, or at all.